docs/source/scala/quickstart.md

```diff
@@ -0,0 +1,159 @@
+
+- To create a new graph 
+
+    ```scala
+    def runWithNewGraph[T](graphID: String = createName, persist: Boolean = false)
+    ```
+    **Note**: The default behaviour is to destory any new graph upon exiting the scope of raphtory context unless chosen to persist explicitly.
+
+- To retreive an existing graph
+
+    ```scala
+    def runWithGraph[T](graphID: String, destroy: Boolean = false)
+    ```
+    **Note**: User can chose to destroy the graph upon exiting the raphtory context by set the `destory` flag to `true`.
+
+- Destroy graphs
+
+    Should user choose to destory any of the persisted graphs for a given graph id
+    ```scala
+    def destroyGraph(graphID: String)
+    ```
+
+These APIs are availabe on `RaphtoryContext` which could be summoned differently for local and remote deployments.
+
+- For Local Deployment
+
+    In order to bring "local" raphtory context in scope, the main application needs to extend `Raphtory.Local` abstract class which requires user to override the `run` method that brings `RaphtoryContext` in scope which could be used to create or retrieve graphs:
+    ```scala
+    object Runner extends RaphtoryApp.Local {
+        override def run(args: Array[String], ctx: RaphtoryContext): Unit = ???
+    }
+    ```
+
+- For Remote Deployment
+
+    In order to bring "remote" raphtory context in scope, the main application should extend `Raphtory.Remote("127.0.0.1", 1736)` abstract class. Since hosts and ports here are already configured as defaults, we need not provide them explicitly:
+    ```scala
+    object RemoteRunner extends RaphtoryApp.Remote() {
+        override def run(args: Array[String], ctx: RaphtoryContext): Unit = ???
+    }
+    ```
+
+## Load Data
+- Should you choose to work with a newly created graph, you must load it with some data:
+    ```scala
+    val path = "/tmp/lotr.csv"
+    val url = "https://raw.githubusercontent.com/Raphtory/Data/main/lotr.csv"
+    FileUtils.curlFile(path, url)
+
+    val source = Source(FileSpout(path), new LotrGraphBuilder())
+    graph.load(source)
+    ```
+
+    Here we download raw data to `/tmp` and then transform it to vertices and edges using graph builders.
+
+- If you are already working with an existing graph, you may not need to load any new data and you can directly jump onto analysis it
+
+
+## Analyse Graph
+Once you have created or retrieved an already existing graph against a `graphId`, you could run analysis on it and choose to write output to a sink of your choice. 
+
+```scala
+graph
+    .execute(Degree())
+    .writeTo(FileSink("/tmp/raphtory"))
+    .waitForJob()
+```
+
+Here, we run `Degree` algorithm on the graph which returns "in-degree" and "out-degree" of nodes in case of a directed graph i.e., incoming or outgoing edges respectively and "degree" of nodes in case of undirected graph representing the total number of edges. Additionally, we chose to write the output to a directory `/tmp/raphtory` using `FileSink`.
```

Comment:
Here, we run `Degree` algorithm on the graph. In the case of a directed graph, the algorithm returns "in-degree" and "out-degree" of nodes - i.e., the incoming or outgoing edges. In the case of an undirected graph, the algorithm returns the "degree" of nodes - representing the total number of edges.